### ISSUE TITLE
How to construct dataset for training?

### ISSUE THREAD

**Issue by oydontbecrazy** [vimalabs/VIMA#19]
Hi, thanks for this amazing work.

Could you provide the interface for loading the training dataset? Is is possible to reveal part of the code for reference? For my understanding, for each task we need to create env as in scripts/example.py, set partition to 'placement_generalization', and task for different 13 tasks and load corresponding data.

**Comment by yunfanjiang** [comment]
Hi there,

Thank you for your interest in our work. You can find more details about training data and a code snippet for data loading [here](https://github.com/vimalabs/VimaBench#training-data).

> For my understanding, for each task we need to create env as in scripts/example.py, set partition to 'placement_generalization', and task for different 13 tasks and load corresponding data.

Please note that evaluation is independent of data loading, therefore you do not need to load any data to run the example. You can refer to [here](https://github.com/vimalabs/VimaBench#create-benchmarking-environments) for instructions on how to create evaluation environments.

Let me know if you have further questions.

**Comment by oydontbecrazy** [comment]
Hi thanks for the respond,

Lets say in def training_step(self, batch, batch_idx), 
May I ask what should the batch contains, please?

Thanks for your kindness. @yunfanjiang

**Comment by yunfanjiang** [comment]
Hi @oydontbecrazy , in our implementation, it contains observation sequence, action sequence, and multimodal prompts. We then perform prompt assembly and tokenization to obtain their embeddings. We then forward our model given these three types of data to compute loss. Masks derived from original sequences are used to mask out loss contributed from padded tokens when we aggregate loss.

**Comment by oydontbecrazy** [comment]
@yunfanjiang Thank you very much for the thoughtful reply. 

So it generally return {
            'prompt': prompt, 
            'orc_actions': action,
            'prompt_assets': prompt_assets,
            } ?

and the obs are initiialzied using self.env.reset().

Then during training, the action token is predicted by gt_history_tokens(gt_obs_segm + gt_obs_action) correct?

Thank you again for being so supportive .

**Comment by oydontbecrazy** [comment]
and by following this, 
I `VimaBench/vima_bench/env/base.py", line 410, in step
    timeout, released = self.task.primitive(VimaBench/vima_bench/env/base.py", line 410, in step
    timeout, released = self.task.primitive(
VimaBench/vima_bench/tasks/components/action_primitives.py", line 203, in __call__
    prepick_pose = utils.multiply(pick_pose, prepick_to_pick)
VimaBench/vima_bench/tasks/utils/misc_utils.py", line 214, in multiply
    return p.multiplyTransforms(pose0[0], pose0[1], pose1[0], pose1[1])
pybullet.error: Invalid input: expected positionA [x,y,z], orientationA [x,y,z,w], positionB, orientationB.
`

All the error occurs in (class PickPlace)

`      pick_pose = (np.array([pose0[0][0], pose0[0][1], 0], dtype=object), pose0[1])
        place_pose = (np.array([pose1[0][0], pose1[0][1], 0], dtype=object), pose1[1])

        # Execute picking primitive.
        prepick_to_pick = ((0, 0, 0.32), (0, 0, 0, 1))
        postpick_to_pick = ((0, 0, self.height), (0, 0, 0, 1))
        prepick_pose = utils.multiply(pick_pose, prepick_to_pick)`

1.  Dim info-pose0: ( torch.size(16,2,3), torch.size(16,2,4) ) , so is the pose 1, 16 is the batch size in your paper.
2. numpy said there is dim error so I added dtype=object
